sdk/nodejs/storage/zMixins.ts

```diff
@@ -397,40 +440,71 @@ export class QueueEventSubscription extends appservice.EventSubscription<QueueCo
 
         const { resourceGroupName, location } = appservice.getResourceGroupNameAndLocation(args, queue.resourceGroupName);
 
-        // The queue binding does not store the storage connection string directly.  Instead, the
-        // connection string is put into the app settings (under whatever key we want). Then, the
-        // .connection property of the binding contains the *name* of that app setting key.
-        const bindingConnectionKey = "BindingConnectionAppSettingsKey";
+        const func = new QueueFunction(name, {
+            ...args,
+            queue
+        });
+
+        super("azure:storage:QueueEventSubscription", name, func, {
+            ...args,
+            resourceGroupName,
+            location,
+        }, opts);
+
+        this.registerOutputs();
+    }
+}
+
+export type QueueFunctionArgs = util.Overwrite<appservice.CallbackArgs<QueueContext, Buffer, void>, {
+    /**
+     * Defines the queue to trigger the function.
+     */
+    queue: Queue;
+}>;
 
-        const bindings: QueueBindingDefinition[] = [{
+/**
+ * Azure Function triggered by a Storage Queue.
+ */
+export class QueueFunction implements appservice.FunctionArgs {

```

Comment:
given that this impls .FunctionArgs, we probably want to call this QueueFunctionArgs.  I think that helps explain what the purpose of this thing is.  It's the args that describe how to build a queue function, and those args will be given to a MultiFunctionApp so it can then actually build the real things.

Reply:
There's actually a thing called `QueueFunctionArgs`: it's the type of the second parameter of the constructor. I like calling them "XyzFunction" - read nice at the client side. Maybe, I'll rename `FunctionArgs` instead.

Reply:
I see.  that's reasonable.  We can say that a Function-App is built up out of Functions.  WFM :)